pkg/credentialprovider/gcp/metadata.go

```diff
@@ -148,18 +166,73 @@ func (g *dockerConfigUrlKeyProvider) Provide() credentialprovider.DockerConfig {
 	return credentialprovider.DockerConfig{}
 }
 
+func runWithBackoff(url string, f func() ([]byte, error)) ([]byte, error) {
+	var backoff = time.Millisecond
+	const maxBackoff = time.Minute
+	for {
+		value, err := f()
+		if err == nil {
+			return value, nil
+		}
+		glog.V(2).Infof("failed to Get %q from gce metadata server: %v", url, err)
```

Comment:
Given that this calls a generic function f() this error message seems awfully specific. Should we log inside of f() if there is an error instead of here?